src/utilhttp.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "utilhttp.h"
+#include "util.h"
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+#include <event2/buffer.h>
+#include <event2/event.h>
+#include <event2/http.h>
+#include <event2/http_struct.h>
+
+// RAII object to clean up after libevent
+struct EventRAII
+{
+    struct event_base *base = nullptr;
+    struct evhttp_connection *http_conn = nullptr;
+    struct evhttp_request *req = nullptr;
+
+    ~EventRAII()
+    {
+        if (http_conn == nullptr && req != nullptr)
+        {
+            // http_conn takes ownership of req, so only release if http_conn
+            // failed to initialize.
+            evhttp_request_free(req);
+            req = nullptr;
+        }
+        if (http_conn != nullptr)
+        {
+            evhttp_connection_free(http_conn);
+            http_conn = nullptr;
+        }
+        if (base != nullptr)
+        {
+            event_base_free(base);
+            base = nullptr;
+        }
+    }
+};
+
+namespace
+{
+static const int CB_UNKNOWN_ERROR = -1;
+}
+
+// The callback function writes the response to this.
+struct CallbackArgs
+{
+    int &http_code;
+    std::stringstream &res;
+};
+
+// Callback for fetching the http response.
+void response_cb(struct evhttp_request *req, void *args_)
+{
+    CallbackArgs *args(static_cast<CallbackArgs *>(args_));
+    if (req == nullptr)
+    {
+        args->http_code = CB_UNKNOWN_ERROR;
+        return;
+    }
+    args->http_code = evhttp_request_get_response_code(req);
+    char buffer[1024];
+    int s = 0;
+    do
+    {
+        assert(static_cast<size_t>(s) <= sizeof buffer);
```

Comment:
I'd like to see a comment describing why this assert should hold.  Something like "evbuffer_remove should never return a value greater than the buffer we send to it."  

Also, please use a DbgAssert and returning CB_UNKNOWN_ERROR if the assertion fails.  I don't think that it makes sense to stop the client in release mode if there's a problem with this http interface.